### PR TITLE
Fix Stakeibc Init/Export Genesis

### PR DESCRIPTION
**Pull request by sampocs**
Closes: #XXX

## Context and purpose of the change
The stakeibc init and export genesis functions had some missing fields, and some unnecessary fields - this PR fixes both functions.

## Brief Changelog
* Added epoch tracker list to stakeibc init and export genesis 
* Removed unnecessary ICAAccount keeper functions
* Removed unnecessary ICAAccount queries
* Removed unnecessary MinValidatorRequirement keeper functions
* Moved epoch tracker key to stakeibc keys
* Removed pagination from epoch tracker query

## Author's Checklist

I have...

- [x] Run and PASSED locally all GAIA integration tests
- [ ] If the change is contentful, I either:
    - [ ] Added a new unit test OR 
    - [ ] Added test cases to existing unit tests
- [ ] OR this change is a trivial rework / code cleanup without any test coverage

If skipped any of the tests above, explain.

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] manually tested (if applicable)
- [ ] confirmed the author wrote unit tests for new logic
- [ ] reviewed documentation exists and is accurate

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes? 
  - [ ] Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`?
  - [ ] This pull request updates existing proto field values (and require a backend and frontend migration)? 
  - [ ] Does this pull request change existing proto field names (and require a frontend migration)?
  How is the feature or change documented? 
      - [ ] not applicable
      - [ ] jira ticket `XXX` 
      - [ ] specification (`x/<module>/spec/`) 
      - [ ] README.md 
      - [ ] not documented

### CODE REVIEW

File: proto/stride/stakeibc/genesis.proto [hunk 1]
```diff
@@ -15,13 +14,10 @@ message GenesisState {
   Params params = 1 [ (gogoproto.nullable) = false ];
   string port_id = 2;
   // list of zones that are registered by the protocol
-  ICAAccount ica_account = 4;
   repeated HostZone host_zone_list = 5 [ (gogoproto.nullable) = false ];
   uint64 host_zone_count = 6;
-  // stores a map from hostZone base denom to hostZone
-  map<string, string> denom_to_host_zone = 9;
```
Comment (shellvish) [inline proto/stride/stakeibc/genesis.proto:22]: thanks for removing this + ica_account!

File: x/stakeibc/genesis.go [hunk 2]
```diff
@@ -10,48 +10,23 @@ import (
...
 	}
-	// Set all the hostZone
-	for _, elem := range genState.HostZoneList {
-		k.SetHostZone(ctx, elem)
+	for _, epochTracker := range genState.EpochTrackerList {
+		k.SetEpochTracker(ctx, epochTracker)
 	}
 
-	// Set hostZone count
-	k.SetHostZoneCount(ctx, genState.HostZoneCount)
```
Comment (shellvish) [inline x/stakeibc/genesis.go:24]: If we're removing HostZoneCount here, I think we should also remove it from the protos. I don't think HostZoneCount is used anywhere in the repo, so this should be safe! 

Also reasonable if we want to split this into another PR given it's not directly related to this issue, but feels like a small change that could be bundled in

File: x/stakeibc/genesis.go [hunk 3]
```diff
@@ -10,48 +10,23 @@ import (
...
-	// genesis.PortId = k.GetPort(ctx)
-	// Get all iCAAccount
-	iCAAccount, found := k.GetICAAccount(ctx)
-	if found {
-		genesis.IcaAccount = &iCAAccount
-	}
+	genesis.Params = k.GetParams(ctx)
+	genesis.HostZoneList = k.GetAllHostZone(ctx)
 	genesis.EpochTrackerList = k.GetAllEpochTracker(ctx)
-	// this line is used by starport scaffolding # genesis/module/export
```
Comment (shellvish) [inline x/stakeibc/genesis.go:54]: Should we remove the scaffolding? I think this implies we can't use `ignite cli` but I don't think we're using it atm regardless

**Review by shellvish — COMMENTED** [review]
Hm, I'm very surprised this doesn't add HostZone to the state when we export, and that the EpochTrackers still aren't being added. I'm going to try testing locally to see if we can export these without an upgrade  

Also really appreciate the standardization of GetEpochTracker on LCD vs CLI endpoints!

File: x/stakeibc/genesis.go (hunk 2 above)
Comment (sampocs) [reply x/stakeibc/genesis.go:24]: I agree! Removing it now

File: x/stakeibc/genesis.go (hunk 3 above)
Comment (sampocs) [reply x/stakeibc/genesis.go:54]: I think we already removed ignite as a dependency so I don't think it's possible to scaffold anymore

But I also think we understand the setup well enough at this point that it's much cleaner to not scaffold!

File: x/stakeibc/keeper/min_validator_requirements.go [hunk 4]
```diff
@@ -1,34 +0,0 @@
-package keeper
```
[the remark this answers is not in the document]
Comment (sampocs) [reply x/stakeibc/keeper/min_validator_requirements.go:1]: yes! good catch

[the remark this answers is not in the document]
Comment (sampocs) [reply x/stakeibc/keeper/min_validator_requirements.go:1]: there's nothing I love more than scrubbing out unused code lol

File: x/stakeibc/genesis.go (hunk 3 above)
Comment (shellvish) [reply x/stakeibc/genesis.go:54]: Agreed!

File: x/stakeibc/keeper/min_validator_requirements.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (shellvish) [reply x/stakeibc/keeper/min_validator_requirements.go:1]: the most addictive drug known to man

**Review by shellvish — APPROVED** [review]
Seems like the issue was in state export on v4.0.3 

Sam and I tested the upgrade flow and local-stride; all seems good!
